lib/friendly_id/finder_methods.rb

```diff
@@ -27,9 +27,20 @@ module FinderMethods
     # @raise ActiveRecord::RecordNotFound
     def find(*args, allow_nil: false)
       id = args.first
-      return super(*args) if args.count != 1 || id.unfriendly_id?
+
+      begin
+        return super(*args) if args.count != 1 || id.unfriendly_id?
+      rescue => e
+        return allow_nil ? nil : raise(e)
+      end
+
       first_by_friendly_id(id).tap { |result| return result unless result.nil? }
-      return super(*args) if potential_primary_key?(id)
+
+      begin
+        return super(*args) if potential_primary_key?(id)
+      rescue => e
+        return allow_nil ? nil : raise(e)
+      end
```

Comment:
I know we talked about exception control flow and I'm definitely not a fan, however I really am not aware of another strategy to use here. I'm open to suggestions though.